frontend/src/Editor/Header/useAppDarkMode.js

```diff
@@ -0,0 +1,54 @@
+import { useMemo } from 'react';
+import { shallow } from 'zustand/shallow';
+import { useEditorStore } from '../../_stores/editorStore';
+import { useAppDataStore } from '../../_stores/appDataStore';
+import { useCurrentStateStore } from '@/_stores/currentStateStore';
+
+const useAppDarkMode = () => {
+  const { appMode, setAppMode } = useEditorStore(
+    (state) => ({
+      appMode: state.appMode,
+      setAppMode: state.actions.setAppMode,
+    }),
+    shallow
+  );
+
+  const { isTJDarkMode } = useAppDataStore(
+    (state) => ({
+      isTJDarkMode: state.isTJDarkMode,
+    }),
+    shallow
+  );
+
+  const { setTheme } = useCurrentStateStore(
+    (state) => ({
+      setTheme: state.actions.setTheme,
+    }),
+    shallow
+  );
```

Comment:
Check once whether the current state updates will re-render this

Reply:
Checked, this is only rendering when there is a render on editor.